### PR TITLE
Refactor v3

### PR DESCRIPTION
**Pull request by edewit**
Goal of this refactor is to make the library more extendable for the client. In order to do that we've introduced a way to create custom "steps". To do so one will have to `extend LauncherStep` and implement:

```javascript   
restoreModel(model: any): void 
```

This is used to save the state in the URL for the authentication redirects.

Usage also changes instead of 'just' adding `<f8launcher ` one needs to add the steps like this:

```html
  <f8launcher (onCancel)="cancel()"> <!-- can remove this if you don't want scrolling wizard layout -->
    <div class="f8launcher-container_nav">
      <f8launcher-step-indicator #stepIndicator></f8launcher-step-indicator>
    </div>
    <div class="f8launcher-container_main">
      <f8launcher-targetenvironment-createapp-step id="TargetEnvironment"> <!-- standalone reusable --> 
      </f8launcher-targetenvironment-createapp-step>
      ...
   </f8launcher>
```
Also new in this version is the ability to customize the 'next buttons' with a section like this:

```html
<f8launcher>
...
  <div class="f8launcher-container_completed-footer">
    <button class="btn btn-primary btn-lg" (click)="complete()">
      View New Application
    </button>
  </div>
</f8launcher>
```
This will also aid in tailored user experience 

BREAKING CHANGE: new client API
fixes: #369

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
> BREAKING CHANGE: new client API

can you point me to those breaking changes so we can understand as impact?

**Comment by edewit** [comment]
@bartoszmajsak instead of using `<f8launcher></f8launcher>` one needs to add the steps that one wants to use for an example of this have a look at the [demo application](https://github.com/fabric8-launcher/ngx-launcher/pull/379/files#diff-4af043d8aae695e36bf185bc6dc8a061). Also the API to send the data to the backend has changed there is no `summary` object anymore but a [_projectile_](https://github.com/fabric8-launcher/ngx-launcher/pull/379/files#diff-0a6e3b1cf4d355947ab94bea5a265c4e) that has to [HttpParams](https://github.com/fabric8-launcher/ngx-launcher/pull/379/files#diff-c0c12181b83511c3afd9b52fa6a739a8R69) function one can use

**Comment by edewit** [comment]
@ia3andy I'll squash on merge?

File: projects/ngx-launcher/src/lib/model/dependency-check.model.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ export class DependencyCheck {
   projectName: string;
   projectVersion: string;
   spacePath: string;
+  targetEnvironment: string;
```
Comment (vikram-raj) [inline projects/ngx-launcher/src/lib/model/dependency-check.model.ts]: ```targetEnviroment``` is used by the only standalone launcher. So, can you make as an optional attribute?

**Comment by vikram-raj** [comment]
@edewit import flow is not working for the demo app. I am getting an error that the repository does not exist.

**Comment by edewit** [comment]
full doc is available https://hackmd.io/wR4tP6KqQZOWqM1Tk3_tUQ?view

**Comment by edewit** [comment]
> @edewit Don't you think we don't need this [variable (nextButtons)](https://github.com/fabric8-launcher/ngx-launcher/pull/379/files#diff-3babff3152da1527475d8d3ef325537aL48) in `ngx-launcher`? I introduce this varibale to handle feature flag for `view pipeline` and `Edit Application in Web IDE` button from `fabric8-ui`.

With the new setup these buttons can be handled by the client (fabric8-ui) no need to have such a switch on the `ngx-launcher` side

**Comment by vikram-raj** [comment]
@edewit So, can you please remove this variable?

**Comment by edewit** [comment]
@vikram-raj oops sorry I thought I did

**Pull request merged**